lib/components-schema-v1_7_x.ts

```diff
@@ -399,45 +441,30 @@ const componentPropertyDefinition: {
         type: 'string',
         description: 'Feature flag that should be present for the property to show up. Always show if not specified.',
     },
-};
-// Define childProperties after the definition of componentPropertyDefinition to avoid
-// a circular reference when creating a copy of componentPropertyDefinition
-// Currently childProperties is only supported one level deep.
-componentPropertyDefinition.childProperties = {
-    type: 'array',
-    description: 'List of conditional child properties',
-    items: {
-        type: 'object',
-        properties: {
-            matchType: {
-                type: 'string',
-                description: `Defines how to match the parent property's value`,
-                enum: ['exact-value'],
-            },
-            matchExpression: {
-                type: ['boolean', 'integer', 'string', 'number'],
-                description: `The expression to use to match the parent property's value`,
-            },
+    childProperties: {
+        type: 'array',
+        description: 'List of conditional child properties',
+        items: {
+            type: 'object',
             properties: {
-                type: 'array',
-                items: {
-                    oneOf: [
-                        {
-                            type: 'string',
-                        },
-                        {
-                            type: 'object',
-                            additionalProperties: false,
-                            // Use a copy to avoid a circular reference and a stack overflow on childProperties at run-time
-                            properties: { ...componentPropertyDefinition },
-                        },
-                    ],
+                matchType: {
+                    type: 'string',
+                    description: `Defines how to match the parent property's value`,
```

Comment:
Property's reads a bit weird. Maybe: "Defines how to match the value of the parent property"?